lib/transaction.js

```diff
@@ -37,6 +37,7 @@ class Transaction extends EventEmitter {
     this.logger = client.logger;
     this.outerTx = outerTx;
     this.trxClient = undefined;
+    this._completed = false;
```

Comment:
Moved this line to earlier in the constructor just to be on the safe side.  (Since the Promise logic is no longer being stalled in the same manner)